lib/plugin/css-variables-collector.js

```diff
@@ -27,10 +27,11 @@ CSSVariablesCollectorPlugin.prototype = {
 		return this.ruleStack.length > 0 && !this.ruleStack[this.ruleStack.length - 1].variable;
 	},
 
-	_isInGlobalOrBaseImport() {
-		return this.config.libName !== "sap.ui.core" && this.importStack.filter((importNode) => {
-			return /\/(?:global|base)\.less$/.test(importNode.importedFilename);
-		}).length > 0;
+	_isVarInLibrary() {
+		// the last less file defines the location of the css variable and this must
+		// be in the path of the current library, otherwise it is an external variable
+		const regexp = new RegExp(`^/resources/${this.config.libPath}/themes/`);
+		return this.importStack.length > 0 ? regexp.test(this.importStack[this.importStack.length - 1].importedFilename) : true;
```

Comment:
FYI, there's already a check that I think has the same goal: https://github.com/SAP/less-openui5/blob/3d16d5a537aaeae26acf0d27bbeda5e7403879ad/lib/plugin/variable-collector.js#L55-L86

Not sure if there's a different requirement in here, but I'd hope that we can use the same logic on both places (or even re-use more).

Reply:
I made the check a bit more simpler 😉